File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        pair = {}

        for i in range(len(nums)):
            if nums[i] in pair and i - pair[nums[i]] <= k:
                return True
            pair[nums[i]] = i

        return False

File: Data Structures & Algorithms/two-integer-sum/submission-1.go
func twoSum(nums []int, target int) []int {
    dMap := make(map[int]int, len(nums))

	for i, n := range nums {
		d := target - n
		if _, ok := dMap[d]; ok {
			return []int{dMap[d], i}
		}
		dMap[n] = i
	}

	return []int{-1, -1}
}
